### PR TITLE
Setup-focused getting started walkthrough

### PR DESCRIPTION
**Pull request by digitarald**
Posting these ideas for the shape of Java's walkthrough. I'd love feedback on this approach and expanding on it further:

Outcome:
- Moving setup steps first avoids popping them up for all users, while ensuring that they come out of the walkthrough with a fully working setup
- Use _when_ clauses

1. Added the _runtime_ and _extension_ setup as first steps.
   - Further potential: Java extension could provide a `when` clause that only shows the _runtime_ step when no JDK was detected
2. Added a `when` clause to only show _open folder_ when workspace doesn't have a folder open.
   - Further potential: Other `when` clauses could further be used for contextual tasks, like detecting Springboard in a project and recommending the extension.

Feel free to try it out. Meanwhile, I am working on recording this to help with feedback.

If we think this direction works well, I would spend more time on polishing the copy.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -121,6 +147,19 @@
               "path": "walkthrough/testing.png",
               "altText": "Testing"
             }
+          },
```
Comment (yaojinyang) [inline package.json:150]: @digitarald  java getting started page will be opened on the top of welcome page? I think better to be opened on the right hand side. also, a good demo for side-by-side view.

Comment (digitarald) [reply package.json:150]: Correct, this is on the list for this release to allow. In my testing, the current Java Tools welcome page does not trigger any side by side views, so I would not block on this.

**Comment by digitarald** [comment]
@yaojinyang @Eskibear recorded a demo of the latest code with a side-by-side onboarding flow:

https://user-images.githubusercontent.com/8599/113538702-c05bc100-9590-11eb-9ad3-49a5af4e0e13.mp4

File: package.json [hunk 2]
```diff
@@ -121,6 +147,19 @@
...
+          {
+            "id": "java.showTesting",
```
Comment (Eskibear) [inline package.json:152]: This task is not shown possibly, because of its duplicated `id`. Is it on purpose?

Comment (digitarald) [reply package.json:152]: No, an oversight from copy/pasta.

**Comment by digitarald** [comment]
@yaojinyang @Eskibear are there any specific concerns or questions that need need to be addressed to move this pull request forward?

**Comment by Eskibear** [comment]
@digitarald Thank you for the rephrasing and nice looking screenshots, it looks much better. I haven't merge this PR because:
* I want to revisit as many use cases as possible, to ensure none of these steps would lead users to a dead end. 
* I was not sure whether the contribution point was finalized. Please let me know if it won't change. 

Besides I don't have other concern to merge it as it's a feature branch. 
Regarding to the newly added markdown support, do we also add multiple actions to better help users? @yaojinyang Or do we polish in incrementally in iterations?

**Comment by yaojinyang** [comment]
> @yaojinyang @Eskibear are there any specific concerns or questions that need need to be addressed to move this pull request forward?
@digitarald I think it's good to go. We can release more improvements through future iterations. Besides @Eskibear's question, I raised another question in our Teams channel, how experience will look like for extensions added post initial VS Code installation.

**Comment by digitarald** [comment]
The updated pull request has the markdown, `toSide` opening, and adds multiple actions for opening and creating a project.

> how experience will look like for extensions added post initial VS Code installation.

A lot can be tweaked here. First step is opening the walkthrough after extension installation; next doing more complex heuristics like opening the walkthrough when the extension activates and no items have been completed yet.

**Review by Eskibear — APPROVED** [review]
Thanks!

**Comment by digitarald** [comment]
@Eskibear thanks for merging! What are your plans for merging https://github.com/microsoft/vscode-java-pack/tree/getting-started-insider into master, so we can start testing the content?

**Comment by Eskibear** [comment]
I can always prepare a private build from this branch for testing. Or do you want me to ship it directly into marketplace?

**Comment by Eskibear** [comment]
@digitarald we'll release a new version this week. As long as it doesn't affect stable version users, I think it's okay for insider users to have a test before we throughly polish these steps/use cases.

**Comment by digitarald** [comment]
> we'll release a new version this week

That would work, as it doesn't do anything for existing users. We could also wait to have an experimentation framework in place, that prevents the build-in Java welcome logic when VS Code has extension walkthroughs enabled.

**Comment by Eskibear** [comment]
> We could also wait to have an experimentation framework in place,

Do you have the timeline when it would be ready?

**Comment by digitarald** [comment]
I'd plan for https://github.com/microsoft/vscode/issues/122592 to land in the next iteration and we can start as soon as extensions adopted it.

**Pull request merged**